crates/uv-toolchain/src/discovery.rs

```diff
@@ -1268,8 +1275,33 @@ impl FromStr for VersionRequest {
     type Err = Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // e.g. `3`, `38`, `312`
+        if let Some(request) = {
+            match s.chars().collect::<Vec<_>>().as_slice() {
```

Comment:
Hmmm this looks odd to me. I suppose it makes the use of pattern matching nicer below, but I'd probably just work in ASCII land here:

```rust
let bytes = s.as_bytes();
if bytes.len() == 1 {
    if (b'0'..=b'9').contains(&bytes[0]) {
        Some(VersionRequest::Major(bytes[0] - b'0'))
    } else {
        None
    }
}
```

And so on for the `MajorMinor`. If you split this into a helper function, you could probably make it even simpler via early returns.